### ISSUE TITLE
Select/dropdown - refine error handling

### ISSUE THREAD

**Issue by mkernohanbc** [bcgov/design-system#388]
Our current implementation (in both Figma and React) indicates an error/invalid state by turning the selector border red:

- Figma: https://www.figma.com/design/Fr33ALEYGvM66CSw2iwxzb/Components?node-id=2518-2519&t=uLivxfoEJXFWKOXO-4
- React: https://design-system-react-components-storybook-ed91fb-dev.apps.silver.devops.gov.bc.ca/?path=/story/components-select-select--invalid

This [isn't sufficient to pass WCAG SC 1.4.1](https://www.w3.org/WAI/WCAG21/Techniques/failures/F81) — we can't _only_ use colour to convey information. 

RAC Select [appears to have built-in support for displaying an error message](https://react-spectrum.adobe.com/react-aria/Select.html#fielderror), but we don't currently appear to be leveraging this.

An invalid selection should be indicated by both:

1. A red border around the selector
2. An error message displayed below the selector

Let's do another pass on this, please: 

- [x] Add error message support to Figma component
- [x] Add error message support to React component

**Update**: [New design in Figma](https://www.figma.com/design/2OhHsH2ztSFUwDcVzxug7Z/BC-Design-System?node-id=3572-4278&t=iltwjpSV8xQQThK4-4)
